gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineNullAway.java

```diff
@@ -134,7 +134,7 @@ public void execute(Configuration _files) {
 
     private static boolean anyProjectUsesJava15(Project proj) {
         return proj.getAllprojects().stream()
-                .anyMatch(project -> project.getTasks().withType(JavaCompile.class).stream()
+                .anyMatch(project -> ImmutableList.copyOf(project.getTasks().withType(JavaCompile.class)).stream()
```

Comment:
I wonder if we would be better off replacing the `Collection.stream()` here using something along these lines:
```java
project.getTasks().withType(JavaCompile.class).matching(comp -> {
  JavaVersion javaVersion = JavaVersion.toVersion(comp.getTargetCompatibility());
  return javaVersion == JavaVersion.VERSION_15;
}).isEmpty()
```

I suspect the TaskContainer may be a bit smarter about modification within iteration when it controls the iteration rather than returning a stream wrapping a spliterator, but it's difficult to say without a reliable reproducer.

Reply:
I tried this by testing `gw classes --stacktrace` on my internal failing repo, and that command:

- fails on `com.palantir.baseline:gradle-baseline-java:4.180.0`
- succeeds on `com.palantir.baseline:gradle-baseline-java:4.180.0-8-ga477298`
- fails on `com.palantir.baseline:gradle-baseline-java:4.180.0-9-g05114a5`

So I think the ImmutableList copy approach works and the TaskCollection one doesn't.  At least it gets these runs to pass.